clippy_lints/src/let_underscore.rs

```diff
@@ -200,7 +200,7 @@ impl<'tcx> LateLintPass<'tcx> for LetUnderscore {
                 }
 
                 // Ignore if it is from a procedural macro...
-                if is_from_proc_macro(cx, init) {
+                if is_from_proc_macro(cx, init) || init.span.from_expansion() {
```

Comment:
Let's check if `local.source` is `Normal` at the top of the function since it would be fine to lint `let _ = some_macro!();`, we can remove the `default_binding_modes` check also in that case

Reply:
Indeed, I hadn't noticed the `source` field. Done.